pgex/examples/platformer_collisions/main.py

```diff
@@ -0,0 +1,125 @@
+"""
+This file is a part of the 'Pygame Examples (pgex)' source code.
+The source code is distributed under the MIT license.
+"""
+
+import asyncio
+
+import pygame
+
+from . import common
+
+
+def get_colliding_tiles(tiles: list, player_rect: pygame.Rect) -> list:
+    """
+    Returns a list of tiles the player is currently colliding with
+    """
+    return_tiles = []
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            return_tiles.append(tile)
+
+    return tiles
+
+
+def calculate_rect(
+    movement: list, player_rect: pygame.Rect, map_tiles: list
+) -> pygame.Rect:
+    """
+    Calculates the Rect of the player based on their movement and the surrounding tiles
+    """
+    player_rect.x += movement[0]
+    tiles = get_colliding_tiles(map_tiles, player_rect)
+    for tile in tiles:
+        if player_rect.colliderect(tile):
+            if movement[0] > 0:
+                player_rect.right = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).left
+            if movement[0] < 0:
+                player_rect.left = pygame.Rect(tile[0], tile[1], tile[2], tile[3]).right
```

Comment:
You are creating the same `Rect` in both `if` statements so I guess you could create it before too avoid redundancy (except you really don't want to create a `Rect` when `movement[0]` is equal to `0`.

Also consider using `elif` structure instead of two separated `if`... No need to check if movement is lower than 0 if we already know that it is in fact greater than 0.